client/state/jetpack-connect/reducer.js

```diff
@@ -182,10 +188,12 @@ export function jetpackConnectAuthorize( state = {}, action ) {
 			);
 		case JETPACK_CONNECT_QUERY_SET:
 			const queryObject = Object.assign( {}, action.queryObject );
+			const shouldAutoAuthorize = ( 'woocommerce-services-auto-authorize' === get( queryObject, 'from' ) );
 			return Object.assign(
 				{},
 				buildDefaultAuthorizeState(),
-				{ queryObject: queryObject }
+				{ queryObject: queryObject },
+				{ autoAuthorize: shouldAutoAuthorize }
```

Comment:
Also, another concern here is that we're overwriting `autoAuthorize` in all cases. Wouldn't this unexpectedly overwrite the existing value in case? For example in the case when creating an account in the JPC flow (auto authorization is supposed to be enabled in that case, but it can get inadvertently disabled when we're not coming from the WooCommerce Services flow). Do we want to be more careful here and don't touch auto authorization if we're outside of the WooCommerce Services flow?